core/src/main/java/org/apache/accumulo/core/clientImpl/Writer.java

```diff
@@ -59,62 +42,18 @@ public Writer(ClientContext context, TableId tableId) {
     this.tableId = tableId;
   }
 
-  private static void updateServer(ClientContext context, Mutation m, KeyExtent extent,
-      HostAndPort server) throws TException, NotServingTabletException,
-      ConstraintViolationException, AccumuloSecurityException {
-    checkArgument(m != null, "m is null");
-    checkArgument(extent != null, "extent is null");
-    checkArgument(server != null, "server is null");
-    checkArgument(context != null, "context is null");
-
-    TabletClientService.Iface client = null;
-    try {
-      client = ThriftUtil.getClient(ThriftClientTypes.TABLET_SERVER, server, context);
-      client.update(TraceUtil.traceInfo(), context.rpcCreds(), extent.toThrift(), m.toThrift(),
-          TDurability.DEFAULT);
-    } catch (ThriftSecurityException e) {
-      throw new AccumuloSecurityException(e.user, e.code);
-    } finally {
-      ThriftUtil.returnClient((TServiceClient) client, context);
-    }
-  }
-
-  public void update(Mutation m) throws AccumuloException, AccumuloSecurityException,
-      ConstraintViolationException, TableNotFoundException {
+  public void update(Mutation m) throws AccumuloException, TableNotFoundException {
     checkArgument(m != null, "m is null");
 
     if (m.size() == 0) {
       throw new IllegalArgumentException("Can not add empty mutations");
     }
 
-    while (true) {
-      TabletLocation tabLoc = TabletLocator.getLocator(context, tableId).locateTablet(context,
-          new Text(m.getRow()), false, true);
-
-      if (tabLoc == null) {
-        log.trace("No tablet location found for row {}", new String(m.getRow(), UTF_8));
-        sleepUninterruptibly(500, MILLISECONDS);
-        continue;
-      }
-
-      final HostAndPort parsedLocation = HostAndPort.fromString(tabLoc.tablet_location);
-      try {
-        updateServer(context, m, tabLoc.tablet_extent, parsedLocation);
-        return;
-      } catch (NotServingTabletException e) {
-        log.trace("Not serving tablet, server = {}", parsedLocation);
-        TabletLocator.getLocator(context, tableId).invalidateCache(tabLoc.tablet_extent);
-      } catch (ConstraintViolationException cve) {
-        log.error("error sending update to {}", parsedLocation, cve);
-        // probably do not need to invalidate cache, but it does not hurt
-        TabletLocator.getLocator(context, tableId).invalidateCache(tabLoc.tablet_extent);
-        throw cve;
-      } catch (TException e) {
-        log.error("error sending update to {}", parsedLocation, e);
-        TabletLocator.getLocator(context, tableId).invalidateCache(tabLoc.tablet_extent);
-      }
+    String table = Ample.DataLevel.of(tableId).metaTable();
 
-      sleepUninterruptibly(500, MILLISECONDS);
+    try (var writer = context.createBatchWriter(table)) {
+      writer.addMutation(m);
+      writer.flush();
```

Comment:
```suggestion
```

Should flush on close anyway. Not much point in calling it manually. The other places this was changed didn't. So, for consistency, I suggest not putting it here either.

Reply:
fixed in a513d75. I think that was vestigial code  (I had a loop or somethnig inside the try block that needed that in past iterations)